CMakeLists.txt

```diff
@@ -1,6 +1,19 @@
 cmake_minimum_required(VERSION 3.17)
 
-project(impylib LANGUAGES C CXX Fortran)
+project(impylib LANGUAGES Fortran)
+
+# Find C/CXX compilers that match the Fortran compiler
+if (CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
+  # Compiler version is something like XY.Z.N, we want only XY
+  string(REGEX MATCH "[0-9]+" COMPILER_MAJOR_VERSION ${CMAKE_Fortran_COMPILER_VERSION})
+  find_program(CMAKE_C_COMPILER "gcc-${COMPILER_MAJOR_VERSION}")
+  find_program(CMAKE_CXX_COMPILER "g++-${COMPILER_MAJOR_VERSION}")
+else()
```

Comment:
What if we need to use clang for C/C++?

Reply:
Then you add some code to handle clang here. This is just for convenience when you use gcc, it does not have to work for everything.